addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlFingerprinter.java

```diff
@@ -44,12 +45,14 @@ public class GraphQlFingerprinter {
             CommonAlertTag.toMap(CommonAlertTag.WSTG_V42_INFO_02_FINGERPRINT_WEB_SERVER);
     private static final Logger LOGGER = LogManager.getLogger(GraphQlFingerprinter.class);
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final BiConsumer<URI, Engine> DEFAULT_APP_CONSUMER = (site, app) -> {};
 
     private final Requestor requestor;
     private final Map<String, HttpMessage> queryCache;
 
     private HttpMessage lastQueryMsg;
     private String matchedString;
+    private static BiConsumer<URI, Engine> appConsumer = DEFAULT_APP_CONSUMER;
```

Comment:
I know I'm late to review but a few notes on the appConsumer...
- The `Engine` (`DiscoveredGraphQlEngine`?) could include the URI 
- The appConsumer (`discoveredEnginesListener`?) could be a functional interface.
- GraphQlFingerprinter could keep a list of listeners instead of a single listener.
- The `raiseFingerPrintingAlert` method can also be added as a listener.

This should make the fingerprinter flexible even for future features that rely on GraphQL engines. That being said I'm good with the PR as-is too and this could even be done in a follow-up PR.

Reply:
I see the vision. I'll tweak for the first item.

I think I'll hold on the listener implementation for another PR. (This one has been dragging I'd like to just get it merged 😇)

Reply:
Isn't a Consumer already a functional interface? Is there a reason to not just simply use a list of Consumers?

If I'm right it wouldn't be too hard to change for this PR.

Reply:
@ricekot I'm pretty sure I'm right the more I think of this.

I can create `List<Consumer> consumers;` Instead of the existing default consumer being noop I could turn it into `raiseFingerPrintingAlert` being the default. If/when the conditions are proper then the Tech Detection consumer can be added. Instead of calling each separately as things currently are I can do something like: `consumers.forEach(consumer -> consumer.accept(discoveredGraphQlEngine);`

Reply:
Sounds good to me.

Reply:
https://github.com/zaproxy/zap-extensions/pull/5843#discussion_r1849367004

There are reasons/benefits to use a custom interface rather than `Consumer` (or any variant of), e.g. if in the future there's the need to add other methods or parameters. If you use `Consumer` you will not be able to do that, with the custom interface you can rely on default methods.

https://github.com/zaproxy/zap-extensions/pull/5843#discussion_r1848883398

Never late :) I was not thinking this would be used by/for other purposes in the future, if that's the intent we should expose the functionality through the extension not as static methods in "internal" classes, proper JavaDoc should be added and so on. (Unless you mean as internal functionality?)

Reply:
Reworked 😀